### PR TITLE
chore: Clean-up, Javadocs, unit tests

### PR DESCRIPTION
**Pull request by maikelarabori**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/GridHeaders.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
 {
     private GridHeaders()
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/GridHeaders.java:46]: we can remove this and annotate `GridHeaders` class with `@NoArgsConstructor(PRIVATE)`

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/SqlQuery.java [hunk 2]
```diff
@@ -58,15 +63,13 @@ public SqlQuery( final String statement, final Map<String, Object> params )
...
     @Override
     public String statement()
     {
         return statement;
     }
 
+    @Override
     public Map<String, Object> params()
     {
         return params;
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/SqlQuery.java:75]: can't we name these methods as regular getters ? In that case we will be able to remove and use `@Getter` annotation

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/DimensionIdentifierConverter.java [hunk 3]
```diff
@@ -48,13 +48,19 @@
...
+     * @param programsAllowed list of programs allowed to be present in
+     *        "fullDimensionId"
+     * @param fullDimensionId string representation of a dimension. Examples:
+     *        abcde[1].fghi[4].jklm, abcde.fghi.jklm, abcde.jklm or jklm
+     * @return the built {@link DimensionIdentifier}
+     * @throws IllegalArgumentException if the programUid in the
+     *         "fullDimensionId" does not belong the list of "programsAllowed"
      */
-    public DimensionIdentifier<Program, ProgramStage, StringUid> fromString( List<Program> programs,
+    public DimensionIdentifier<Program, ProgramStage, StringUid> fromString( List<Program> programsAllowed,
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/DimensionIdentifierConverter.java:63]: `allowedPrograms` sounds better

**Review by gnespolino — APPROVED** [review]
This was a great job, thank you.
Left some minor comments

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/GridHeaders.java (hunk 1 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/GridHeaders.java:46]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/SqlQuery.java (hunk 2 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/SqlQuery.java:75]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/DimensionIdentifierConverter.java (hunk 3 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/processing/DimensionIdentifierConverter.java:63]: Done

**Pull request merged**